lib/NetworkManager.js

```diff
@@ -151,25 +170,36 @@ class NetworkManager extends EventEmitter {
       }).catch(debugError);
     }
 
-    if (event.redirectUrl) {
-      const request = this._interceptionIdToRequest.get(event.interceptionId);
-      if (request) {
-        this._handleRequestRedirect(request, event.responseStatusCode, event.responseHeaders, false /* fromDiskCache */, false /* fromServiceWorker */, null /* securityDetails */);
-        this._handleRequestStart(request._requestId, event.interceptionId, event.redirectUrl, event.isNavigationRequest, event.resourceType, event.request, event.frameId, request._redirectChain);
-      }
-      return;
-    }
     const requestHash = generateRequestHash(event.request);
     const requestId = this._requestHashToRequestIds.firstValue(requestHash);
     if (requestId) {
+      const requestWillBeSentEvent = this._requestIdToRequestWillBeSentEvent.get(requestId);
+      this._onRequest(requestWillBeSentEvent, event.interceptionId);
       this._requestHashToRequestIds.delete(requestHash, requestId);
-      this._handleRequestStart(requestId, event.interceptionId, event.request.url, event.isNavigationRequest, event.resourceType, event.request, event.frameId, []);
+      this._requestIdToRequestWillBeSentEvent.delete(requestId);
     } else {
       this._requestHashToInterceptionIds.set(requestHash, event.interceptionId);
-      this._handleRequestStart(null, event.interceptionId, event.request.url, event.isNavigationRequest, event.resourceType, event.request, event.frameId, []);
     }
   }
 
+  /**
+   * @param {!Protocol.Network.requestWillBeSentPayload} event
+   * @param {?string} interceptionId
+   */
+  _onRequest(event, interceptionId) {
+    let redirectChain = [];
+    if (event.redirectResponse) {
+      const request = this._requestIdToRequest.get(event.requestId);
+      // If we connect late to the target, we could have missed the requestWillBeSent event.
```

Comment:
This comment is suspicious now.

Reply:
It's still valid: when we attach to the page after the initial request but before the redirect.